src/main/java/net/openhft/chronicle/core/io/VanillaReferenceCounted.java

```diff
@@ -158,9 +162,9 @@ public void throwExceptionIfNotReleased() throws IllegalStateException {
 
     @Override
     public void warnAndReleaseIfNotReleased() throws ClosedIllegalStateException {
-        if (refCount() > 0) {
+        if (valueGetAndSet(0) > 0) {
             if (!unmonitored && !AbstractCloseable.DISABLE_DISCARD_WARNING)
-                Slf4jExceptionHandler.WARN.on(type, "Discarded without being released");
+                Jvm.warn().on(type, "Discarded without being released");
             callOnRelease();
```

Comment:
We do atomic get-and-set to ensure we were the ones to take the value to zero and it's safe to `callOnRelease`